python_modules/dagster/dagster/core/storage/runs/migration.py

```diff
@@ -138,3 +140,17 @@ def add_run_stats(run_storage: RunStorage, run_id: str) -> None:
                 end_time=run_stats.end_time,
             )
         )
+
+
+def migrate_run_repo_tags(run_storage: RunStorage, print_fn=None):
+    from dagster.core.storage.runs.sql_run_storage import SqlRunStorage
+
+    if not isinstance(run_storage, SqlRunStorage):
+        return
+
+    if print_fn:
+        print_fn("Querying run storage.")
+
+    for run in chunked_run_iterator(run_storage, print_fn):
+        if run.external_pipeline_origin:
+            run_storage.add_run_tags(run.run_id, run.tags_for_storage())
```

Comment:
is this idempotent if the tags are already there for some reason (e.g. a failure partway through)? if not should it be?